pynest/nest/__init__.py

```diff
@@ -399,6 +405,19 @@ def __dir__(self):
     )
 
 
+def _setattr_error(self, attr, val):
```

Comment:
@Helveg If I understand the code here right, this checks if there is a class attribute of the given name and raises an error only if that is not the case. Could you add a comment describing the mechanism here, so we remember it five years from now?